src/box/memtx_tree.cc

```diff
@@ -772,32 +772,44 @@ prepare_start_prefix_iterator(struct memtx_tree_key_data<USE_HINT> *start_data,
 	return true;
 }
 
-template <bool USE_HINT>
-static int
-tree_iterator_start(struct iterator *iterator, struct tuple **ret)
+/**
+ * Creates an iterator based on the given key, after data and iterator type.
+ * If the iterator creation is not finished and requires a step back, the
+ * @a step_back is set to true (false otherwise).
+ *
+ * Also updates @a start_data and iterator @a type as required.
+ *
+ * @param tree - the tree to lookup in;
+ * @param start_data - the key to lookup with;
+ * @param after_data - the after key, can be empty if not required;
+ * @param type - the lookup iterator type;
+ * @param region - the region to allocate a new @a start_data on if required.
+ * @param[out] iterator - the result of the primary lookup;
+ * @param[out] equals - true if the primary lookup gave the exact key match;
+ * @param[out] step_back - true if the result iterator is not valid and needs
+ *  to be adjusted using the memtx_tree_iterator_prev to reach the target.
```

Comment:
I don't like that the caller may need to "step back" after calling this function. AFAIU the function doesn't step back by itself for the caller to get the successor. IMO it's better to return the successor instead of the flag.

Reply:
Done.